flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/FlinkOperator.java

```diff
@@ -190,7 +189,8 @@ void registerSessionJobController() {
         var metricManager =
                 MetricManager.createFlinkSessionJobMetricManager(baseConfig, metricGroup);
         var statusRecorder = StatusRecorder.create(client, metricManager, listeners);
-        var reconciler = new SessionJobReconciler(eventRecorder, statusRecorder);
+        var autoscaler = AutoscalerFactory.create(client, eventRecorder);
+        var reconciler = new SessionJobReconciler(eventRecorder, statusRecorder, autoscaler);
```

Comment:
Session Job doesn't use the autoscaler before, and autoscaler will be used after this PR, right?

Reply:
Yes, seemed like an unnecessary limitation. We can also include this during our testing .